datumaro/components/merge/exact_merge.py

```diff
@@ -215,25 +221,33 @@ def _merge_point_clouds(self, item_a: DatasetItem, item_b: DatasetItem) -> Point
         media = None
 
         if isinstance(item_a.media, PointCloud) and isinstance(item_b.media, PointCloud):
-            if item_a.media.path and item_b.media.path and item_a.media.path != item_b.media.path:
+            item_a_path = getattr(item_a.media, "path", None)
+            item_b_path = getattr(item_b.media, "path", None)
+
+            if item_a_path and item_b_path and item_a_path != item_b_path:
                 raise MismatchingMediaPathError(
-                    (item_a.id, item_a.subset), item_a.media.path, item_b.media.path
+                    (item_a.id, item_a.subset), item_a_path, item_b_path
                 )
 
-            if item_a.media.path or item_a.media.extra_images:
+            # Avoid direct comparison here for better performance
+            # If there are 2 "data-only" pointclouds, they won't be compared and
+            # we just use the first one
+            if item_a.media.has_data or item_a.media.extra_images:
                 media = item_a.media
 
                 if item_b.media.extra_images:
                     for image in item_b.media.extra_images:
                         if image not in media.extra_images:
                             media.extra_images.append(image)
-            else:
+            elif item_b.media.has_data or item_b.media.extra_images:
                 media = item_b.media
 
                 if item_a.media.extra_images:
                     for image in item_a.media.extra_images:
                         if image not in media.extra_images:
                             media.extra_images.append(image)
+            else:
+                assert False, "Unknown cloudpoint field combination"
```

Comment:
Please raise some exception rather than `assert`.

Reply:
Fixed in https://github.com/openvinotoolkit/datumaro/pull/944/commits/c62b17ee374d5f30046b52073d4bf9ba34ac53e0